robot/basestation/static/js/elements/navigation/navigation.js

```diff
@@ -1,191 +1,545 @@
+                $('#goal-DMS-min-input.long.goal-' + current).attr("value", mins)
+                $('#goal-DMS-sec-input.long.goal-' + current).attr("value", secs)
+            } else {
+                throw "You didn't enter anything for long"
+            }
+
+            let goalElem = $('#goal-name.goal-' + current)
+            let goalName = goalElem.val()
+            if (goalName.length != 0) goalElem.attr('value', goalName)
+            else goalElem.attr('value', 'Goal-' + current)
+
+            $('#goal-name-fieldset.goal-' + current).prop('disabled', true)
+            $('#goal-lat-fieldset.goal-' + current).prop('disabled', true)
+            $('#goal-long-fieldset.goal-' + current).prop('disabled', true)
+
+
+            /*----------------------------------\
+            this section implements ROS goal node
+            \----------------------------------*/
+            let goalData = new ROSLIB.Message({
+                name: goalName,
+                long: long,
+                lat: lat
+            })
+
+            create_goal_publisher.publish(goalData)
+
+        } catch (e) {
+            $('#goal-confirm-btn.goal-' + current).prop('disabled', false)
+            $('#goal-change-btn.goal-' + current).prop('disabled', true)
+
+            rosLog('ROSWARN', e)
+        }
+    }
+
+    function goalHandlerTemplate(mode, format, current) {
+        $("#goal-" + mode + "-" + format + "-btn.goal-" + current).on('click', function(event) {
+            event.preventDefault()
+
+            $('#goal-' + mode + '-fieldset.goal-' + current).attr("format", format)
+            $('#goal-' + mode + '-select-format.goal-' + current).dropdown('toggle')
+            $('#goal-' + mode + '-select-format.goal-' + current).detach()
+
+            let inputTemplate = $('#goal-' + format + '-input-template').html()
+            $('#goal-' + mode + '-input-group.goal-' + current).append(inputTemplate)
+            $('#goal-' + mode + '-input-group.goal-' + current).find('input').addClass('goal-' + current + ' ' + mode)
+            $('#goal-' + mode + '-input-group.goal-' + current + ' span.input-group-text:first').text(mode)
+        })
+    }
+
+    function createGoalLatitudeHandler(current) {
+        goalHandlerTemplate('lat', 'DD', current)
+        goalHandlerTemplate('lat', 'DDM', current)
+        goalHandlerTemplate('lat', 'DMS', current)
+    }
+
+    function createGoalLongitudeHandler(current) {
+        goalHandlerTemplate('long', 'DD', current)
+        goalHandlerTemplate('long', 'DDM', current)
+        goalHandlerTemplate('long', 'DMS', current)
+    }
+
+    function createGoalButtons(current) {
+        let goalButtons = $("#goal-buttons").html()
+        $("#goal-buttons-input-group.goal-" + current).append(goalButtons)
+        $("#goal-buttons-input-group.goal-" + current).find('*').addClass('goal-' + current)
+
+        createGoalConfirmButtonHandler(current)
+        //createGoalChangeButtonHandler(current)
```

Comment:
This should either be deleted or uncommented

Reply:
I decided to delete it, as it wasn't necessary.